### PR TITLE
ENH: cupyx/scipy/interpolate: add make_lsq_spline

### PR DESCRIPTION
**Pull request by ev-br**
cross-ref https://github.com/cupy/cupy/issues/7186

Add `make_lsq_spline`, lay the groundwork for FITPACK translations (splrep/splprep and *UnivariateSpline classes). Those will come is a follow-up PR, to keep the diff sizes manageable.

Algorithmically, this is solving a linear least-squares problem with banded matrices. Since banded linear algebra is not available from CUDA toolkit, we add a specialized kernel. As discussed offline, the kernel is essentially sequential.
(also note that even in cuSolver, a possibly helpful function, `csrlsqvqr` is also only available for CPU only).

### CODE REVIEW

File: cupyx/scipy/interpolate/_bspline2.py [hunk 1]
```diff
@@ -529,3 +528,345 @@ def _make_periodic_spline(x, y, t, k, axis):
...
+            }
+        }
+        if (i < nc) {
+            offset[i] = i;
+        }
+
+    } // for(i = ...
+}
+
+'''
```
Comment (takagi) [inline cupyx/scipy/interpolate/_bspline2.py:704]: Just suppressing the flake8 check.
```suggestion
'''  # NOQA
```

**Review by takagi — DISMISSED** [review]

**Comment by ev-br** [comment]
Merged as a part of https://github.com/cupy/cupy/pull/8267, closing.

**Pull request closed without merging**